cns/restserver/ipam_test.go

```diff
@@ -1529,3 +1532,124 @@ func TestIPAMFailToRequestPartialIPsInPool(t *testing.T) {
 		t.Fatalf("Expected fail requesting IPs due to only having one in the ipconfig map, IPs in the pool will not be assigned")
 	}
 }
+
+func TestIPAMGetSWIFTv2IPSuccess(t *testing.T) {
+	svc := getTestService()
+	middleware := middlewares.NewMockSWIFTv2Middleware()
+	svc.AttachSWIFTv2Middleware(middleware)
+
+	middleware.SetEnvVar()
+	defer middleware.UnsetEnvVar() //nolint:errcheck // ignore error
+	middleware.SetMTPNCReady()
+
+	ncStates := []ncState{
+		{
+			ncID: testNCID,
+			ips: []string{
+				testIP1,
+			},
+		},
+		{
+			ncID: testNCIDv6,
+			ips: []string{
+				testIP1v6,
+			},
+		},
+	}
+
+	// Add Available Pod IP to state
+	for i := range ncStates {
+		ipconfigs := make(map[string]cns.IPConfigurationStatus, 0)
+		state := NewPodState(ncStates[i].ips[0], ipIDs[i][0], ncStates[i].ncID, types.Available, 0)
+		ipconfigs[state.ID] = state
+		err := UpdatePodIPConfigState(t, svc, ipconfigs, ncStates[i].ncID)
+		if err != nil {
+			t.Fatalf("Expected to not fail adding IPs to state: %+v", err)
+		}
+	}
+
+	req := cns.IPConfigsRequest{
+		PodInterfaceID:   testPod1Info.InterfaceID(),
+		InfraContainerID: testPod1Info.InfraContainerID(),
+	}
+	b, _ := testPod1Info.OrchestratorContext()
+	req.OrchestratorContext = b
+	req.DesiredIPAddresses = make([]string, 2)
+	req.DesiredIPAddresses[0] = testIP1
+	req.DesiredIPAddresses[1] = testIP1v6
+
+	resp, err := svc.requestIPConfigHandlerHelper(context.TODO(), req)
+	if err != nil {
+		t.Fatalf("Expected to not fail requesting IPs: %+v", err)
+	}
+	podIPInfo := resp.PodIPInfo
+
+	if len(podIPInfo) != 3 {
+		t.Fatalf("Expected to get 3 pod IP info (IPv4, IPv6, Multitenant IP), actual %d", len(podIPInfo))
+	}
+
+	// Asserting that SWIFT v2 IP is returned
+	assert.Equal(t, SWIFTv2IP, podIPInfo[2].PodIPConfig.IPAddress)
+	assert.Equal(t, SWIFTv2MAC, podIPInfo[2].MacAddress)
+	assert.Equal(t, cns.DelegatedVMNIC, podIPInfo[2].NICType)
+	assert.False(t, podIPInfo[2].SkipDefaultRoutes)
+}
+
+func TestIPAMGetSWIFTv2IPFailure(t *testing.T) {
```

Comment:
Not sure what we are asserting here and what is failing, can you please explain with some comments at the beginning?